src/opnsense/mvc/app/models/OPNsense/Interfaces/VxLan.php

```diff
@@ -28,8 +28,56 @@
 
 namespace OPNsense\Interfaces;
 
+use Phalcon\Messages\Message;
 use OPNsense\Base\BaseModel;
 
 class VxLan extends BaseModel
 {
+    public function performValidation($validateFullModel = false)
+    {
+        $messages = parent::performValidation($validateFullModel);
+
+        foreach ($this->vxlan->iterateItems() as $vxlan) {
```

Comment:
here comes the tricky part in terms of what to validate. When `$validateFullModel` is set to `true`, everything should be handled, but the default is to only validate what has changed. That's why in most of our code you see constructions like these:

```
foreach ($this->getFlatNodes() as $key => $node) {
    if ($validateFullModel || $node->isFieldChanged()) {
       // XXX : some logic
    }
```

In this case, since our validation only covers 3 fields, you could also consider to traverse using the `iterateItems()` as you're already doing, but just check if any of the related fields has changed. Something like 

`$validateFullModel || $vxlan->vxlangroup->isFieldChanged() || ...`  When that's the case, it's safe to proceed.

When always enforcing validation, someone with two broken records in the set would not be able to fix his issue anymore due to being locked (on screen is one record, the other one will also report being broken),. Just try it out and you see what I mean.

Reply:
Oh now I understand. I did test it now. I added 3 broken records, afterwards I added the validation and I tried to create a new entry. But I couldn't since the whole model was validated, and not only the changed records. Thanks! Now I know what I have to do. :)